src/unix/tty.c

```diff
@@ -62,7 +64,18 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
    * other processes.
    */
   if (type == UV_TTY) {
-    r = uv__open_cloexec("/dev/tty", O_RDWR);
+    /* Reopening a pty in master mode won't work either because the reopened
+     * pty will be in slave mode (*BSD) or reopening will allocate a new
+     * master/slave pair (Linux)
+     *
+     * This ptsname call should return NULL if this fd isn't a pty in master
+     * mode. If it succeeds the pty is in master mode and reopening should be skipped.
+     */
+    if (ptsname(fd) == NULL && ttyname_r(fd, path, sizeof(path)) == 0) {
```

Comment:
I don't think this issue applies here as even if the global buffer is changed concurrently, the check does not care about the actual value of the buffer. It only checks if the pointer to that buffer is not NULL.

If you think it's an issue nevertheless I'll change it accordingly.